pkgs/development/compilers/paco/default.nix

```diff
@@ -1,18 +1,29 @@
-{ lib, buildGoPackage, fetchFromGitHub }:
+{ lib, buildGoModule, fetchFromGitHub, fetchpatch }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "paco";
   version = "1.0";
 
   src = fetchFromGitHub {
     owner = "pacolang";
     repo = "paco";
     rev = "v${version}";
-    sha256 = "03x75h40dhjswbf2g1408krj9b1w05y9pjzygzhklldc75r3n9dh";
+    hash = "sha256-sCU7cjmsUTrhf/7Lm3wBPKwk80SAhCfc4lrCBggspw8=";
   };
 
-  goPackagePath = "github.com/pacolang/paco";
-  goDeps = ./deps.nix;
+  vendorHash = "sha256-J0TKp1df5IWq3Irlzf1lvhWlXnP//MsVqs9M8TtEraw=";
+
+  patches = [
+    # Set correct package path in go.mod
+    (fetchpatch {
+      url = "https://github.com/pacolang/paco/pull/1/commits/886f0407e94418d34c7e062c6857834aea3c99ac.patch";
```

Comment:
Fetching from a pull request doesn't seem right...

Reply:
Ya, I don't think it makes sense to do this. If the tool is unmaintained but still used, the people that use it can fork and run with it - nixpkgs isn't the right place to maintain a fork.

I am also not sure it makes sense to move off of `buildGoPackage` for these.. Is there a specific reason for it? If so, it should be linked into the PR.

Reply:
As long as we don't end up on https://patch-diff.githubusercontent.com/ via e.g. ``https://github.com/pacolang/paco/pull/1.patch`` it should be fine.

Reply:
I see, thanks for the info!

@aaronjheng could you include a link to the upstream pull request for these `buildGoModule` changes you are doing? It would be helpful for reviewers, especially when the patch doesn't match the repo (not the case for this specific PR, but one or two of the others you have submitted were for different module names or repos - which is confusing at best.. suspicious at worst)

Reply:
Thanks for the clarification. @SuperSandro2000. 

> one or two of the others you have submitted were for different module names or repos

@qbit Nice catch! I assume you mean https://github.com/NixOS/nixpkgs/pull/212593. The canonical repo has been transfered from `gchaincl/httplab` to `qustavo/httplab`. I will fix it latter.

> could you include a link to the upstream pull request for these buildGoModule changes you are doing

Patches about Go Modules support in my rencent PRs are all from upstream repo if I recall correctly. I suppose I should have provided a PR link in addition to the commit url as you said. Thanks!

Reply:
Yep, that's the one! Thank you! :D